src/top.rs

```diff
@@ -155,7 +155,15 @@ enum State {
     Healthy,
 
     /// The link was explicitly partitioned.
-    ExplicitPartition,
```

Comment:
I haven't seen a need for this yet, but one idea to achieve this is to model a link with two direction states and allow setting either one. This also might make the A,B -> src,dst stuff cleaner. wdyt?

Reply:
I think it would be a lot cleaner if the 'links'-map in the 'Topology'-struct didn't deduplicate links. I.e, if the network topology between any two nodes 'a' and 'b' was described by two links, 'a'->'b' and 'b'->'a', instead of having one multidirectional link.

However, that feels like a rather big change, and it might also have other downsides.

We could just split 'state' in struct 'Link' into 'state_a_to_b' and 'state_b_to_a'. I worry that the tricky A,B->src,dst stuff would basically remain the same though?

Reply:
I agree we should hold off on a larger refactor. We've been discussing a larger restructuring of the network to support different ip versions and a more realistic lo for some time now. We can tackle that then. 

This idea doesn't avoid complexity around the directions, but it does make it really clear what is happening in each direction. I think its worth doing that as part of this change.

```rust
struct Link {
    a_b: State,
    b_a: State,
}
```

Reply:
Right! I made an attempt, and I have to say it was a big improvement. Thanks for the idea!